### DISCUSSION TITLE
Quick Question About Color Settings in ESPHome AwoX BLE Mesh Hub

### DISCUSSION THREAD

**Discussion by rewisch** [fsaris/EspHome-AwoX-BLE-mesh-hub#85] [Q&A] [answered]
Hey there,

First off, big thanks for the awesome ESPHome component: AwoX BLE mesh (MQTT) hub. I've got it all set up and running in ioBroker, and I'm having a blast controlling my lamps and playing around with the colors.

But I've hit a bit of a snag. Even though I can tweak the RGB codes to create some cool scenes, I can't seem to replicate the simple "light" settings I can do in the app, like switching between warm or cool light. I noticed there are RGB and brightness modes, but messing with those hasn't quite worked out for me.

Any tips or pointers on where I might look to get this sorted?

Here's what I'm working with:

product_id: 0x4B
device_type: RGB
name: EGLO Ceiling
model: ECeiling_30
manufacturer: EGLO

I'm controlling the bulbs like this:

`setState('mqtt.1.awox-ble-mesh-hub.3496.command', JSON.stringify(payload));`

`let payload = {
    state: "ON",
    color_mode: "rgb",
    brightness: brightness,
    color: {
        r: r,
        g: g,
        b: b
    }
};`

Thanks a ton for the help and for the great project!

Cheers,
Reto

**Comment by fsaris** [comment, marked as answer]
Hi,

you don't need to send all the info.

if you want to set the color temp use:
```
{ "color_temp": 248}
```

or to change the brightness
```
{ "brightness": 79}
```
or change color
```
{ "color": { "r": 255, "g": 74, "b": 61 }}
```

You can combine `color` and `brigtness` or `color_temp` and `brightness`. You don't need to pass the color_mode. That is something used by HomeAssistant internallySee https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/blob/b21e6fbf78ba65d404fab6e917aa9a11dbad7633/components/awox_mesh/awox_mesh_mqtt.cpp#L540 for how the send parameters are handled by the code.

So to switch between light modes you need to send the "previous" `color_temp` or `color` rgb values.

Groet Frans

**Reply by rewisch** [reply]
ah, heureka! thank you very very much!
Have a great weekend. 
Greetings Reto
